### ISSUE TITLE
Linux e2e tests on GitHub are now mostly failing

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#1150]
See an example here: https://github.com/kiwix/kiwix-js/actions/runs/6839494846/job/18598176146#step:9:101. It's happening pretty regularly, almost always on the same spot: Edge on Linux with the language selector in the Gutenberg test ZIM. It seems we don't wait long enough for the element to be ready, as the error (when trying to change ZIM language) is:

`ElementNotInteractableError: element not interactable: Element is not currently visible and may not be manipulated`

**Comment by Jaifroid** [comment]
Issue now occurring almost every time. It's an issue with Microsoft Edge only on Linux. However, when I run the same test locally in WSL with Edge installed, it passes every time. The main difference is that when running locally, the app resets itself. This does dismiss a dialogue box which latest Edge shows on start -- could have something to do with it (as we don't reset when doing it on GitHub).

**Comment by Jaifroid** [comment]
So, @Rishabhg71, it may be coincidence, but it seems that forcing an app reset before running the Gutenberg spec fixes this issue. See the run here: https://github.com/kiwix/kiwix-js/actions/runs/6840275468/job/18599365522?pr=1153, and the experimental commit https://github.com/kiwix/kiwix-js/pull/1153/commits/c9c6f17b1cbc8d8afc567a9415d1df27d9017ff7 (which I'll shortly revert, as it doesn't belong in that PR).

This seems to be due to the latest version of Edge having a feature notification UI element displayed on startup, and it's dismissed by enforcing a reset. The only part of the tests affected by this seems to be the language selection box in Gutenberg main page.

Forcing a reset has the disadvantage of adding about 7-11 seconds PER Gutenberg suite test (running jQuery and SW modes, but it only resets once at the beginning, not again after switching modes).

Possibilities are:

* Find another way to fix this, e.g., a simple reload (`window.location.reload()`) might be enough to dismiss the dialogue box, not so drastic as a reset;
* Omit the Edge test for Linux on GitHub (as we already test it on Windows) -- however, I'm not sure this really fixes the issue, and should only be used if other methods don't work.

**Comment by Jaifroid** [comment]
Hmm, maybe it was just luck, because I've just had a failure with the reset fix still in my PR. See:

https://github.com/kiwix/kiwix-js/actions/runs/6840718362/job/18600294095

This would point more towards the test not waiting long enough for the element to become active. But it's curious that only Edge on Linux is affected.

**Comment by Rishabhg71** [comment]
I will start working on this in the next few hours

**Comment by Jaifroid** [comment]
I can pretty much confirm that what I wrote above in https://github.com/kiwix/kiwix-js/issues/1150#issuecomment-1807097893 was just coincidence. It's still failing just as much, even with that "fix".

**Comment by Jaifroid** [comment]
Fun fact: when I "watch" the Linux test, it passes. When I don't, it fails. Maybe it's running on a quantum computer: observing it causes the wave function to collapse.... 😛

Seriously, though, it's possible that tests run a bit slower when connected to the console and reporting to a tab observed by a human. This could make the difference in adding enough latency for the language selector in the Gutenberg ZIM to be found. (Sheer guesswork.)

**Comment by Jaifroid** [comment]
Just to add to this, having experimentally disabled the Linux Edge test on one of the PRs, I'm now hitting issues with the BrowserStack Firefox 70 test, possibly due to the dialogue box showing in the screenshot below. It never gets dismissed between the JQuery and the ServiceWorker tests. @Rishabhg71 Do you remember whether you ever found a way to dismiss that, or did it not matter? My issue with this test in #1164 might be something else.

![image](https://github.com/kiwix/kiwix-js/assets/4304337/d13599bc-6b20-4156-a42e-611f40c989f9)

**Comment by Rishabhg71** [comment]
No this is the first time asking for a dialog When I was working on this test it didn't ask for any prompt. This could be easily fixed by adding a CLI arg when launching the browser (I don't remember the exact CLI arg)

**Comment by Jaifroid** [comment]
> No this is the first time asking for a dialog When I was working on this test it didn't ask for any prompt. This could be easily fixed by adding a CLI arg when launching the browser (I don't remember the exact CLI arg)

OK, thanks. In any case, I've determined that my issue has nothing to do with this dialogue box ☹️. I've made a PR #1165 that addresses the issues on main (but not on my PR, which needs further work), by disabling Edge testing on Linux (because we already test Edge Chromium on Windows fine, and also because we test Chrome on Linux, which is pretty much the same thing, with no issues). It also cleans up a few other things. I'll ask you to take a look at that and approve (or review).
